File: marketplaces/service.go
// Copyright (C) 2021 Creditor Corp. Group.
// See LICENSE for copying information.

package marketplaces

import (
	"context"

	"github.com/google/uuid"
)

// Service is handling marketplaces related logic.
//
// architecture: Service
type Service struct {
	marketplaces DB
}

// NewService is a constructor for marketplaces service.
func NewService(marketplaces DB) *Service {
	return &Service{
		marketplaces: marketplaces,
	}
}

// Create add lot in DB.
func (service *Service) Create(ctx context.Context, lot Lot) error {
	return service.marketplaces.Create(ctx, lot)
}

// Get returns lot by id from DB.
func (service *Service) Get(ctx context.Context, id uuid.UUID) (Lot, error) {
	return service.marketplaces.Get(ctx, id)
}

// ListActive returns active lots from DB.
func (service *Service) ListActive(ctx context.Context) ([]Lot, error) {
	return service.marketplaces.ListActive(ctx)
}

File: web/console/src/marketplace/service.ts
// Copyright (C) 2021 Creditor Corp. Group.
// See LICENSE for copying information.

import { MarketplaceClient } from '@/api/marketplace';
import { Lot, MarketPlacePage } from '@/marketplace';
import { CreatedLot } from '@/app/types/marketplace';

/**
 * exposes all arketplace domain entity related logic
 */
export class Marketplaces {
    protected readonly marketplace: MarketplaceClient;

    /** default marketplaceClient implementation */
    public constructor(marketplace: MarketplaceClient) {
        this.marketplace = marketplace;
    };

    /** returns marketplace domain entity with list of lots */
    public async list(selectedPage: number): Promise<MarketPlacePage> {
        return await this.marketplace.list(selectedPage);
    };

    /** creates lot */
    public async createLot(lot: CreatedLot): Promise<void> {
        await this.marketplace.createLot(lot);
    };

    /** returns lot by id */
    public async getLotById(id: string): Promise<Lot> {
        return await this.marketplace.getLotById(id);
    };
};
